src/shapes/text.class.js

```diff
@@ -879,6 +887,27 @@
       graphemeInfo.angle = info.angle + (this.pathSide ===  'right' ? Math.PI : 0);
     },
 
+    _getGraphemeDirection: function (grapheme, lineIndex, charIndex) {
+      if (this._reExplicitRTL.test(grapheme)) {
+        return 'rtl';
+      }
+      else if (this._reExplicitLTR.test(grapheme)) {
+        return 'ltr';
+      }
+      else if (lineIndex === 0 && charIndex === 0) {
+        return this.direction === 'auto' ?
+          fabric.util.getElementStyle(fabric.document.body, 'dir') || 'ltr' :
+          this.direction;
+      }
+      else if (charIndex === 0) {
+        var prevLine = this.__charBounds[lineIndex - 1];
+        return prevLine[prevLine.length - 1].dir;
+      }
+      else {
+        return this.__charBounds[lineIndex][charIndex - 1].dir;
```

Comment:
O wonder if this is the desired logic for spaces

Reply:
i can ask to people that are expert about this.
But in general unless we find evidence, a space is one of the nor left nor right chars, so it behaves like the others that are such, it look at the context.

Reply:
solved